pennylane_lightning/core/src/simulators/lightning_qubit/utils/SparseLinAlg.hpp

```diff
@@ -38,23 +71,48 @@ namespace Pennylane::LightningQubit::Util {
  * @return result       result of the matrix vector multiplication.
  */
 template <class fp_precision, class index_type>
-std::vector<std::complex<fp_precision>>
-apply_Sparse_Matrix(const std::complex<fp_precision> *vector_ptr,
-                    const index_type vector_size, const index_type *row_map_ptr,
-                    [[maybe_unused]] const index_type row_map_size,
-                    const index_type *entries_ptr,
-                    const std::complex<fp_precision> *values_ptr,
-                    [[maybe_unused]] const index_type numNNZ) {
-    std::vector<std::complex<fp_precision>> result;
-    result.resize(vector_size);
-    std::size_t count = 0;
-    for (index_type i = 0; i < vector_size; i++) {
-        result[i] = 0.0;
-        for (index_type j = 0; j < row_map_ptr[i + 1] - row_map_ptr[i]; j++) {
-            result[i] += values_ptr[count] * vector_ptr[entries_ptr[count]];
-            count++;
+std::vector<std::complex<fp_precision>> apply_Sparse_Matrix(
+    const std::complex<fp_precision> *vector_ptr, const index_type vector_size,
+    const index_type *row_map_ptr,
+    [[maybe_unused]] const index_type row_map_size,
+    const index_type *entries_ptr, const std::complex<fp_precision> *values_ptr,
+    [[maybe_unused]] const index_type numNNZ, index_type num_threads = 0) {
+    // Output vector initialized to zero
+    std::vector<std::complex<fp_precision>> result(
+        vector_size, std::complex<fp_precision>(0.0));
+
+    // Determine the number of threads to use
+    if (num_threads <= 0) {
+        const int max_threads = std::thread::hardware_concurrency();
+        num_threads =
+            std::min(vector_size, static_cast<index_type>(max_threads));
+    }
+
+    // Divide the rows approximately evenly among the threads
+    index_type chunk_size = (vector_size + num_threads - 1) / num_threads;
+    std::vector<std::thread> threads;
```

Comment:
We could rewrite line 93 "std::vector<std::thread> threads;" as "std::vector<std::jthread> threads;" in order to use jthread. It would simplify the thread management since jthread doesnt require explicite use of join on the different threads after execution. We could also use std::views::iota(start, end) from the std::range library when calling emplace_back later in the code, it would helps avoid manually handling the iteration indices and make the code easier to read. Also, if later we would change the chunk logic, using ranges we won't have to rewrite the loop logic.